core/src/main/scala/org/apache/spark/deploy/history/config.scala

```diff
@@ -64,4 +64,11 @@ private[spark] object config {
       .bytesConf(ByteUnit.BYTE)
       .createWithDefaultString("1m")
 
+  val IN_PROGRESS_ABSOLUTE_LENGTH_CHECK =
+    ConfigBuilder("spark.history.fs.inProgressAbsoluteLengthCheck.enabled")
```

Comment:
Is there any disadvantage in just leaving this always on? Otherwise this doesn't need to be configurable.

Reply:
How much overhead are we talking about?

That thread is not really in any performance critical path, and in general people won't have so many running apps that this should become a problem... but that kinda depends on how bad this new call is.

Reply:
This call makes two invocations, one for getting blocks info from the NameNode and another for getting the last block info from DataNode. I agree this is not a performance critical path, I will remove the config in the update.